### PR TITLE
DNS resolution proposal in Hotpot

### PR DESCRIPTION
**Pull request by leonwanghui**

### CODE REVIEW

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+## Motivation
+
+As we know, DNS resolution is widely used to service-to-service communication.
+For example, when users want to access Google Search, they would type some
+DNS addresses such like `www.google.com` rather than some static IP address.
+
+Currently there are several services deployed in OpenSDS cluster: api-server,
+controller, dock, dashboard, authchecker and database. And the current design is
+that system admins have to remember all their IP addresses and configure them
+in `opensds.conf`. This could lead to several problems:
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: Add a reference github link to this file.

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+* Quite difficult to scale-out. Multi-dock deployment could drive users to crazy
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: drive users crazy (remove "to")

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+### Hotpot related
+In Hotpot source code, we need to add the new term of `DnsEndpoint` to decouple
+bind IP address with external IP address.
+
+Here are some terminology to help developers understand this design:
+* ApiEndpoint: indicates the bind endpoint which the server would listen to. The
+format of this field should be exactly like `127.0.0.1:8088`. All Hotpot internal
+services should include it and also it should be configured properly.
+* DnsEndpoint: indicates the external IP and endpoint which the server would
+expose to other services. Normally the `DnsEdnpoint` should be the same with
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: the same as

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+`ApiEndpoint`, but in K8S condition it should be the format like `FQDN`.
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: Remove "condition".

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+### K8S related
+In this design, some manifests (see [here](https://github.com/opensds/opensds/tree/18c9088053bd99aa363182cbc582e2e232361a74/install/kubernetes)) are added under `install/kubernetes` folder, and `README.md` provides a tutorial
+how to install OpenSDS on an existing Kubernetes cluster.
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: on how to ...

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+[osdsdock]
+api_endpoint = 0.0.0.0:50050
++dns_endpoint = localhost:50050
+# Choose the type of dock resource, only support 'provisioner' and 'attacher'.
+dock_type = provisioner
+# Specify which backends should be enabled, sample,ceph,cinder,lvm and so on.
+enabled_backends = sample
+
+[database]
+-credential = opensds:password@127.0.0.1:3306/dbname
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: What is replacing the credential?

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+and [#27](https://github.com/opensds/design-specs/pull/27)) need to learn how
+this design works, and then update their current implementation. 
+
+## Use Cases
+
+### OpenSDS Deployment on Kubernetes
+In Kubernetes, all applications would be deployed and managed in `Pod`, and user
+can access to that application through `Service`. So if OpenSDS wants to take
+the advantages of K8S orchestration, it has to distinguish the concept of `PodIP`
+with `ClusterIP`. For detailed introduction of K8S Service, please refer to
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: s/with/from

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+-endpoint = localhost:2379,localhost:2380
++dns_endpoint = localhost:2379,localhost:2380
+driver = etcd
+```
+
+### Developer impact
+
+Owners of these designs ([#28](https://github.com/opensds/design-specs/pull/28)
+and [#27](https://github.com/opensds/design-specs/pull/27)) need to learn how
+this design works, and then update their current implementation.
```
Comment (xing-yang) [inline specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: Please add a link to this spec in those open design-specs.

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md (hunk 6 above)
Comment (leonwanghui) [reply specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: credential is only required for mysql database, so leave it here seems a bit confusing.

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md (hunk 8 above)
Comment (leonwanghui) [reply specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: Sure

File: specs/capri/DNS_Resolution_Proposal_in_Hotpot.md (hunk 1 above)
Comment (leonwanghui) [reply specs/capri/DNS_Resolution_Proposal_in_Hotpot.md]: Done

**Comment by xing-yang** [comment]
@wisererik Please take a look.

**Pull request merged**